src/Test/Utilities/Portable/Diagnostics/DiagnosticDescription.cs

```diff
@@ -219,13 +217,8 @@ public override bool Equals(object obj)
                 {
                     if (_startPosition.Value != d._startPosition.Value)
                     {
-                        _showPosition = true;
-                        d._showPosition = true;
```

Comment:
Sounds like good RoslynAnalyzer candidate to check if fields are set in .Equals, .GetHashCode, .ToString... methods